fix(github-actions): refine sensitive file check

iamnikhilranjan opened the pull request:
- Converted SENSITIVE_FILES to a JSON array in the workflow for easier management.
- Improved triggers to run checks on new commits, updates, and re-runs.
- Used jq for reliable parsing of sensitive file patterns.
- Preserved all existing sensitive file patterns.
- Made the workflow more consistent, secure, and maintainable.

The implementation follows the robust solution already implemented in talawa-admin ([#4292](https://github.com/PalisadoesFoundation/talawa-admin/issues/4292)) and addresses all issues mentioned in #1024.

 <!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**

a bugfix

**Issue Number:**

Fixes #1024 

**Did you add tests for your changes?**

No

**Snapshots/Videos:**

N/A

**If relevant, did you update the documentation?**

No

**Summary**

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**

<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-docs/blob/develop/CONTRIBUTING.md)?**

<!--Yes or No-->

iamnikhilranjan committed:
fix(github-actions): refine sensitive file check
- Use grep -E and merge-base for stability
- Update patterns and error format
- Align with Talawa Admin
Fixes #1024

iamnikhilranjan commented:
Hi, Please add the 'ignore-sensitive-files-pr' label.

palisadoes commented on .github/workflows/pull-request.yml:
```diff
@@ -131,7 +128,10 @@ jobs:
         id: changed-files
         run: |
-          FILES_COUNT=$(git diff --name-only --diff-filter=ACMRT ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | wc -l)
+          # Get the base branch ref
```
1. This section must not be changed.
2. It is entirely out of scope of the issue

iamnikhilranjan replied:
My apologies, I’ll revert this part and fix the PR as per the issue.

palisadoes reviewed (changes requested):
This does not reflect the very clearly stated requirements in the issue. 
Please re-read it and apply the appropriate changes.

iamnikhilranjan committed:
feat: Make sensitive files GitHub Action more robust

- Convert SENSITIVE_FILES from bash array to proper YAML JSON list format
- Add workflow trigger types to support re-runs (opened, synchronize, reopened)
- Update SENSITIVE_PATTERNS variable to match repository needs
- Use jq for robust JSON parsing of sensitive patterns
- Maintain all existing sensitive file patterns for talawa-docs

Fixes #1024

palisadoes commented:
1. This PR needs to be closed. You are not following the example in the issue.
2. This is a relatively simple task.
3. I'm going to unassign you and give it to someone else.
